src/ToEnum.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\Filter;
+
+use BackedEnum;
+use Laminas\Filter\Exception\RuntimeException;
+use Laminas\Stdlib\ArrayUtils;
+use Traversable;
+use UnitEnum;
+
+use function is_array;
+use function is_int;
+use function is_string;
+use function is_subclass_of;
+
+/**
+ * @psalm-type Options array{enum: class-string<UnitEnum>}
```

Comment:
I wonder if we should keep extending `AbstractFilter` at all, but that's something @gsteel should help us decide.

In this case, `new ToEnum(MyEnum::class)` seems cleaner to me, and doesn't really need the API inherited from `AbstractFilter`, although unsure if divergence from that is better or worse.

Adding `AbstractFilter` can also be done afterwards too.

Reply:
But that would not work with the factory? Or should I add a factory to handle the case? 
Extending AbstractFilter  could be easily removed because I do not really need it.